File: MACombinedTextField/MACombinedTextField.m
//
//  WASCustomTextField.m
//  MumsPlayground
//
//  Created by MadApper on 23/06/2013.
//  Copyright (c) 2013 We Are Social. All rights reserved.
//

#import "MACombinedTextField.h"

@implementation MACombinedTextField

- (id)initWithFrame:(CGRect)frame
{
    frame = CGRectMake(frame.origin.x, frame.origin.y, frame.size.width, MAX(frame.size.height, 40));
    self = [super initWithFrame:frame];
    if (self) {

        self.label = [[UILabel alloc]initWithFrame:CGRectMake(5, 0, frame.size.width/5*2-5, 0)];
        self.label.backgroundColor = [UIColor clearColor];
        self.label.textColor = [UIColor lightGrayColor];
        self.label.userInteractionEnabled = false;
        self.label.adjustsFontSizeToFitWidth = true;
        [self addSubview:self.label];
        
        
        self.textField = [[UITextField alloc]initWithFrame:CGRectMake(self.label.frame.size.width+5, 0, self.frame.size.width/5*3-5, 0)];
        self.textField.textColor = [UIColor whiteColor];
        [self addSubview:self.textField];
        
        self.layer.backgroundColor = [[[UIColor whiteColor]colorWithAlphaComponent:0.1]CGColor];
        self.layer.borderColor = [[[UIColor whiteColor]colorWithAlphaComponent:0.2]CGColor];
        self.layer.borderWidth = 1;
        self.layer.cornerRadius = 3;
        
        divider = [[UIView alloc]initWithFrame:CGRectMake(self.label.frame.size.width, 5, 1, frame.size.height-10)];
        divider.backgroundColor = [[UIColor whiteColor]colorWithAlphaComponent:0.1];
        [self addSubview:divider];
        
        [self setLabelText:@"" withLabelFont:[UIFont systemFontOfSize:[UIFont systemFontSize]] withTextFieldFont:[UIFont systemFontOfSize:[UIFont systemFontSize]]];
        [self setBorderColor:[UIColor lightGrayColor]];
        
    }
    return self;
}

-(void)setLabelText:(NSString *)text withLabelFont:(UIFont *)lfont withTextFieldFont:(UIFont *)tfont{
    
    CGSize size = [text sizeWithAttributes:@{NSFontAttributeName:lfont}];
    self.label.frame = CGRectMake(self.label.frame.origin.x, self.frame.size.height/2-size.height/2, self.label.frame.size.width, size.height);
    self.label.text = text;
    self.label.font = lfont;
    
    self.textField.frame = CGRectMake(self.textField.frame.origin.x, self.frame.size.height/2-size.height/2, self.textField.frame.size.width, size.height);
    
    self.textField.font = tfont;
}

-(void)setBorderColor:(UIColor *)color{
    divider.backgroundColor = color;
    
    self.layer.borderColor = color.CGColor;
}


@end

File: MACombinedTextField/MACombinedTextField.swift
//
//  MACombinedTextField.swift
//  MACombinedTextFieldSwiftDemo
//
//  Created by Paul Napier on 26/06/2014.
//  Copyright (c) 2014 madapper. All rights reserved.
//

import Foundation
import UIKit

class MACombinedTextField:UIView{
    
    let divider = UIView()
    let label = UILabel()
    let textField = UITextField()
    
    init(frame: CGRect) {
        super.init(frame:CGRect(x:frame.origin.x,y:frame.origin.y,width:frame.size.width,height:max(frame.size.height,40)))
        label.frame = CGRect(x:5, y:0, width:frame.size.width/5*2-5, height:0)
        label.backgroundColor = UIColor.clearColor()
        label.textColor = UIColor.lightGrayColor()
        label.userInteractionEnabled = false
        label.adjustsFontSizeToFitWidth = true
        addSubview(label)
        
        textField.frame = CGRect(x:self.label.frame.size.width+5, y:0, width:self.frame.size.width/5*3-5, height:0)
        textField.textColor = UIColor.whiteColor()
        addSubview(textField)
        
        divider.frame = CGRect(x:self.label.frame.size.width, y:5, width:1, height:frame.size.height-10)
        addSubview(divider)
        
        layer.backgroundColor = UIColor.whiteColor().colorWithAlphaComponent(0.1).CGColor
        layer.borderWidth = 1;
        layer.cornerRadius = 3;
        
        setLabelText("", labelFont: UIFont.systemFontOfSize(UIFont.systemFontSize()), textFieldFont: UIFont.systemFontOfSize(UIFont.systemFontSize()))
        setBorderColor(UIColor.lightGrayColor())
    }
    
    func setLabelText(text:NSString, labelFont:UIFont, textFieldFont:UIFont){
        var size = text.sizeWithAttributes([NSFontAttributeName:labelFont])
        label.frame = CGRect(x:label.frame.origin.x, y:frame.size.height/2-size.height/2, width:label.frame.size.width, height:size.height)
        label.text = text
        label.font = labelFont
        
        textField.frame = CGRect(x:textField.frame.origin.x, y:frame.size.height/2-size.height/2, width:textField.frame.size.width, height:size.height)
        textField.font = textFieldFont
    }
    
    func setBorderColor(color:UIColor){
        divider.backgroundColor = color
        layer.borderColor = color.CGColor
    }
    
}


